### PR TITLE
support nested generics

### PR DESCRIPTION
**Pull request by turowicz**
When the T was generic we didn't see what was really being submitted.

### CODE REVIEW

File: Ve.Metrics.StatsDClient.Tests/CastleWindsor/StatsDCountingInterceptorShould.cs [hunk 1]
```diff
@@ -37,6 +37,13 @@ public void It_should_count_the_targeted_generic_method_and_log_to_statsd()
         }
 
         [Test]
+        public void It_should_count_the_targeted_nested_generic_method_and_log_to_statsd()
+        {
+            Service.TrackedGenericMethod<IEnumerable<object>>();
+            StatsdMock.Verify(x => x.LogCount("dependencies.ienumerable`1-object", It.IsAny<int>(), It.IsAny<Dictionary<string, string>>()), Times.Once);
```
Comment (andyroyle) [inline Ve.Metrics.StatsDClient.Tests/CastleWindsor/StatsDCountingInterceptorShould.cs:43]: whats with the backticks?

File: Ve.Metrics.StatsDClient.CastleWindsor/StatsDTimingInterceptor.cs [hunk 2]
```diff
@@ -28,10 +30,17 @@ protected override void Invoke(IInvocation invocation, StatsDTiming attr)
...
+                var generic = new List<string>();
                 var arguments = methodBase.GetGenericArguments();
-                var generic = arguments[0].Name;
 
-                name = name.Replace("{generic}", generic.ToLowerInvariant());
+                while (arguments.Any())
+                {
+                    var argument = arguments.First();
+                    generic.Add(argument.Name.ToLowerInvariant());
+                    arguments = argument.GetGenericArguments();
```
Comment (andyroyle) [inline Ve.Metrics.StatsDClient.CastleWindsor/StatsDTimingInterceptor.cs]: can you wrap some of this in a method to make it more obvious what it's doing?

Just checking i've read this right, it's searching down through the sets until it hits a concrete class?

File: Ve.Metrics.StatsDClient.Tests/CastleWindsor/StatsDCountingInterceptorShould.cs (hunk 1 above)
Comment (turowicz) [inline Ve.Metrics.StatsDClient.Tests/CastleWindsor/StatsDCountingInterceptorShould.cs:43]: They are added by .NET when you get the type name and are removed by statsD when submitting.

Comment (turowicz) [inline Ve.Metrics.StatsDClient.Tests/CastleWindsor/StatsDCountingInterceptorShould.cs:43]: If we remove backticks in .NET we will be duplicating the behaviour.

File: Ve.Metrics.StatsDClient.CastleWindsor/StatsDTimingInterceptor.cs (hunk 2 above)
Comment (turowicz) [inline Ve.Metrics.StatsDClient.CastleWindsor/StatsDTimingInterceptor.cs]: on it

Comment (turowicz) [inline Ve.Metrics.StatsDClient.CastleWindsor/StatsDTimingInterceptor.cs]: done

File: Ve.Metrics.StatsDClient.Tests/CastleWindsor/StatsDCountingInterceptorShould.cs (hunk 1 above)
Comment (andyroyle) [inline Ve.Metrics.StatsDClient.Tests/CastleWindsor/StatsDCountingInterceptorShould.cs:43]: 👍

**Pull request merged**